### PR TITLE
Added AnimeSaturn provider

### PR DESCRIPTION
**Pull request by winstriker**
Added AnimeSaturn.cc italian provider

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        val title = document.select("img.cover-anime").first()!!.attr("alt")
+        val japTitle = document.select("div.box-trasparente-alternativo").first()!!.text()
+        val posterUrl = document.select("img.cover-anime[src]").first()!!.attr("src")
+        var malId : Int? = null
+        var aniListId : Int? = null
+
+        document.select("[rel=\"noopener noreferrer\"]").forEach {
+            if(it.attr("href").contains("myanimelist"))
+                malId = it.attr("href").removeSuffix("/").split('/').last().toInt()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:100]: ToIntOrNull

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            else
+                aniListId = it.attr("href").removeSuffix("/").split('/').last().toInt()
+        }
+
+        val plot = document.select("div#shown-trama").first()!!.text()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:105]: ?.

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+        val tags = ArrayList<String>()
+        document.select("a.generi-as").forEach {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:108]: = .map { it. text()}

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            tags.add(it.text())
+        }
+
+        val details : List<String> = document.select("div.container:contains(Stato: )").first()!!.text().split(" ")
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:112]: ?.

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        var status : String? = null
+        var duration : String? = null
+        var year : String? = null
+        var score : String? = null
+
+        val isDubbed = document.select("div.anime-title-as").first()!!.text().contains("(ITA)")
+
+        details.forEach {
+            when (it){
+                "Stato:" -> status = details[details.indexOf(it)+1]
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:122]: move out indexof for dry code

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+                "episodi:" -> duration = details[details.indexOf(it)+1]
+                "uscita:" -> year = details[details.indexOf(it) + 3]
+                "Voto:" -> score = details[details.indexOf(it)+1].split("/")[0]
+                else -> return@forEach
+            }
+        }
+
+        val episodes = ArrayList<Episode>()
+
+        document.select("a.bottone-ep").forEach{
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:132]: episodes = .mapnotnull

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+                    data = it.attr("href"),
+                    episode = episode.toInt()
+                )
+            )
+        }
+
+        return newAnimeLoadResponse(title, url, TvType.Anime) {
+            this.engName = title
+            this.japName = japTitle
+            this.year = year!!.toInt()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimeSaturnProvider.kt:150]: ?. and tointornull

**Review by LagradOst — COMMENTED** [review]
Minor changes, doing !! instead of ?. will cause nullptr exeption when not found

**Comment by winstriker** [comment]
Done! Thanks for your fixes!

**Comment by LagradOst** [comment]
LGTM, ready to merge?

**Comment by winstriker** [comment]
Yes 👍

**Pull request merged**